File: Stack/leetcode/225.go
// Package leetcode provides 
package leetcode

type MyStack struct {
	queue []int
}


func Constructor() MyStack {
	return MyStack{
			queue: make([]int, 0),
	}
}


func (this *MyStack) Push(x int)  {
	this.queue = append(this.queue, x)
}


func (this *MyStack) Pop() int {
	n := len(this.queue) - 1
	for n != 0 {
			val := this.queue[0]
			this.queue = this.queue[1:]
			this.queue = append(this.queue, val)
			n--
	}

	val := this.queue[0]
	this.queue = this.queue[1:]
	return val
}


func (this *MyStack) Top() int {
	val := this.Pop()
	this.queue = append(this.queue, val)
	return val
}


func (this *MyStack) Empty() bool {
	return len(this.queue) == 0
}

File: Queue/cpp/225.cpp
#include <queue>
using namespace std;

class MyStack {
  queue<int> que;
  MyStack() {}

  void push(int x) {
    que.push(x);
  }

  int pop() {
    int size = que.size();
    --size;
    if (size--) {
      que.push(que.front());
      que.pop();
    }
    int result = que.front();
    que.pop();
    return result;
  }

  int push() {
    return que.back();
  }

  bool empty() { return que.empty(); }
};
